apps/core/views.py

```diff
@@ -293,3 +295,33 @@ def post(self, request, *args, **kwargs):
         user.save()
 
         return Response('Your password has been changed')
+
+
+class PasswordChangeView(APIView):
+    queryset = User.objects.all()
+    permission_classes = (IsAuthenticated,)
+    serializer_class = PasswordChangeSerializer
+
+    def get_serializer(self, *args, **kwargs):
+        return self.serializer_class(self, *args, **kwargs)
+
+    @swagger_auto_schema(
+        tags=['auth'],
+    )
+    def post(self, request, *args, **kwargs):
+        user = request.user
+        serializer = self.get_serializer(
+            data=request.data,
+            context={"user": user}
+        )
+
+        if not serializer.is_valid(raise_exception=True):
+            return Response(
+                {"detail": "Password is invalid"},
```

Comment:
We have to use the error data from the serialiser

Reply:
I found that i don't need to check is_valid in bool operations at all, because serialisers will raise ValidationErrors if not valid.
can i leave it like this?
```python
    def post(self, request, *args, **kwargs):
        user = request.user
        serializer = self.get_serializer(
            data=request.data,
            context={"user": user}
        )
        serializer.is_valid(raise_exception=True)
        user.set_password(serializer.validated_data["password"])
        user.save()

        return Response('Your password has been changed')
```

Reply:
No, if you know an error is likely to occur, you must handle it correctly

Reply:
is that better?
```python
        if not serializer.is_valid(raise_exception=True):
            return Response(
                serializer.errors,
                status=status.HTTP_400_BAD_REQUEST,
            )
```